### DISCUSSION TITLE
Precompiled releases with the filament sensor signal reversed.

### DISCUSSION THREAD

**Discussion by Amor-Roma** [Jyers/Marlin#1074] [Ideas]
I have tried cheap switches, as well and the BIGTREETECH Smart Filament Sensor, and both give me a filament runout error right at the start of the print.  I even tried to reverse the wires as many tutorials instruct, but then it doesn't seem to work at all.  It would be nice to have a release that has the filament runout sensor detection level signal be high, and when no material, the material port level signal goes low.  This seems to be the way majority of the sensors work, in fact I haven't been able to find one that doesn't work that way.

**Comment by smooreace** [comment]
Compile your own. what your wanting takes every bit of a few seconds

**Comment by Cal-LF** [comment]
Learn to compile and you can make other changes as needed also. This has been discussed in other threads and found that signal level low is the more common. Making a precompile for each would double the number of releases and is not practical.
I have to compile my own for the use of BTT motion filament sensor. It is set to low and 10mm.
It is not difficult to learn how to use VS Code. Follow the many You Tube videos to get started.
It would be nice to be able to have a menu select for high or low but is determined when compiling.

**Reply by Amor-Roma** [reply]
I was just thinking it would be practical since having many releases with different pin points is practical.  If there can be build with 5x5, 10x10, 15x15, and that is practical, I don't see how adding 1 more build for each with a reversed signal would be much different.

**Reply by Cal-LF** [reply]
@Amor-Roma Wow! Compared to you learning to compile your own, adding 28 more percompiled releases is asking a bit much. Pretty sure that is not going to happen.

**Reply by Jyers** [reply]
While it would be dead simple to add this option to the menu, it would require changing how marlin handles the runout state. Making it non-constant. Unfortunately such a change would never make it upstream. Therefore this most likely will not be implemented in this firmware, as I do hope at some point to contribute my work to the main Marlin repo

**Comment by Cal-LF** [comment]
One of the primary so-called features of the E3V2 is having a filament runout sensor. That was one of the Creality's selling points when the printer first released although Creality firmware does not support it. I think it caught the eye of many buyers including me. 
Having filament runout fully implemented should include the high/low state menu check box if possible. It is a recurring request.

**Comment by smooreace** [comment]
Of course not, youre not the one that's doing anything. Invest a little
time in yourself and learn to compile your own.
